### ISSUE TITLE
[MAC] Game won't ready up after updates

### ISSUE THREAD

**Issue by mmuldoon** [ValveSoftware/Dota-2#745]
After the January Dota 2 update I started receiving a lot more abandons from the game not connecting to the server after the matchmaking screen. I've been penalized 2 times now with 24 hour holds on playing dota because for some reason the game stalls out on the loading screens. This never happened to me before.

So I'm now sitting with 19hrs left on my most recent abandon because of an update from Dota. I was thinking of purchasing the Winter Pass but after these two issues it might be time to find a new game to spend my money on. Help would be awesome, removing this 19hr hold would be even more awesome.

**Comment by gdrewb-valve** [comment]
This is unlikely to be a Mac issue, I'll pass it on to the larger team.

**Comment by mmuldoon** [comment]
Thanks Drew,
Do you know is there anyway to get my account hold off?

**Comment by ravzski** [comment]
Me too. I have been penalized so many times because it crashes on the loading screen after the accept game stage.

**Comment by gdrewb-valve** [comment]
Sorry, this is just a specific non-Windows tech support repo, you'll want to post in dev.dota2.com about possible account adjustment.

**Comment by Multisaft** [comment]
Its nice to see that other mac os x dota2 players have the same problem as I am encountering. 

I dont like to crosspost but from another topic here at github:

I'am running a Mac Pro with an GTX 970 TI with the actual 10.10.5, security update 2016-001 as well as the actual nvidia web drivers and I have the exact same problem like you have, even that I don't hear any cliks in the background, dota2 just crashes every single time when the match is loading. I have this in ranked as well as all pick games, single drafts etc. **What I am able to load is the 10on10 mod without any issues and without any crashes!!!** Can you check that out for me?

Even worse, after todays steam update I dont even get a crash report that I can show you. Dota2 just stop minimising to desktop again. That suxxx!!

Plz Valve, help us to fix this problem. Any Feedback is highly welcome!

**Comment by Tele42** [comment]
@Multisaft please open a separate issue report, posting randomly on other issue reports is hard to track.

**Comment by Multisaft** [comment]
Hi Tele42, Ive dont what you have asked me for.

**Comment by gdrewb-valve** [comment]
@mmuldoon (or anybody with this issue) can you post your console logs?  I know you're risking more bans if you try it again so no worries if you don't want to bother but they might help us figure out what's going on.

**Comment by gdrewb-valve** [comment]
@mmuldoon have you ever gotten a message about being unable to connect to a region?  If you're doing Find Match/Select Region do you see reasonable ping times?

**Comment by mmuldoon** [comment]
@gdrewb-valve I've never gotten a unable to connect to region, ping rates for the US are 50-70ms, other countries pings are 150+. I'll also post my console logs if you tell me how to do it. Thanks Drew!!

**Comment by mmuldoon** [comment]
Here ya go Drew, this was a failed connection, cost me a 19min ban but hopefully it helps solve this issue. Let me know if you need anything else to help out.

```
Allocating new transform texture!
Writing configuration for slot 0
Sending ad-hoc CMsgClientPingData to GC.
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(backgrounds/accept_popup_test) no such group
Saving lobby data to disk.
WriteSteamRemoteStorageFileAsync( "scripts/dota_local_disk_lobby_data.txt" ) -> at 181.550
Telling Steam not to update the app for the next 900 seconds
WriteSteamRemoteStorageFileAsync( "cfg/last_match.dat" ) -> at 181.553
Detected lobby in state RUN, connecting to =[A:1:4230360073:6362]
SwitchToLoop remoteconnect requested:  id [2] addons []
CAsyncWriteInProgress::OnComplete( "scripts/dota_local_disk_lobby_data.txt" ) -> Success at 182.420
CAsyncWriteInProgress::OnComplete( "cfg/last_match.dat" ) -> Success at 182.422
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(backgrounds/accept_popup_test) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(backgrounds/accept_popup_test) no such group
CL:  IGameSystem::LoopDeactivateAllSystems
WriteSteamRemoteStorageFileAsync( "scripts/control_groups.txt" ) -> at 182.777
CL:  Disconnecting from server: #GameUI_Disconnect_LoopDeactivate
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(backgrounds/dashboard_parallax_test) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/campaign_winter2016_header) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/ui_tier_max) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/find_match_status) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/play_searching) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/home_button) no such group
SV:  IGameSystem::LoopDeactivateAllSystems
HO:  IGameSystem::LoopDeactivateAllSystems
SV:  Disconnect client 'I TROLL HARD' from server(1): NETWORK_DISCONNECT_SHUTDOWN
WriteSteamRemoteStorageFileAsync( "voice_ban.dt" ) -> at 182.915
Network socket 'client' opened on port -1
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( remoteconnect : success )
CAsyncWriteInProgress::OnComplete( "scripts/control_groups.txt" ) -> Success at 182.935
CAsyncWriteInProgress::OnComplete( "voice_ban.dt" ) -> Success at 182.935
CL:  Sending connect to =[A:1:4230360073:6362]
Attempting connection to service in data center 'iad'
Requesting session from ord#9 (162.254.193.70:27023).  Ping = 39+14=53 (front+back=total).
Selected ord#9 (162.254.193.70:27023) as primary, session 0881620d.  [Ping = 39+14=53 (front+back=total).]
Requesting session from iad#28 (162.254.192.70:27042).  Ping = 60+0=60 (front+back=total).
Selecting iad#28 (162.254.192.70:27042) as backup #1, session 36eb04e1.  Ping = 60+0=60 (front+back=total).
CL:  Sending connect to =[A:1:4230360073:6362]
CL:  Received S2C_CHALLENGE [799530920 auth 3] from =[A:1:4230360073:6362]
CL:  Sending C2S_CONNECT [43 protocol 799530920 auth 3] to =[A:1:4230360073:6362]
CL:  Received S2C_CONNECTION from =[A:1:4230360073:6362] [addons:'']
CL:  Connected to '=[A:1:4230360073:6362]'
CL:  Suppress INetchannel::Transmit() in loopmode( remoteconnect )
CL:  CLoopModeRemoteConnect::OnClientFrameSimulate switching to "levelload" loopmode with addons:  
SwitchToLoop levelload requested:  id [2] addons []
Failed to load image for cursor from resource/cursor/workshop/necronub/cursor_inivisible.bmp: Couldn't open resource/cursor/workshop/necronub/cursor_inivisible.bmp
ChangeGameUIState: DOTA_GAME_UI_STATE_DASHBOARD -> DOTA_GAME_UI_STATE_LOADING_SCREEN
**** Panel MapPreview has fill-parent-flow for width, but isn't in a flowing right layout
**** Panel MapPreview has fill-parent-flow for width, but isn't in a flowing right layout
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( levelload : success )
CL:  Permit INetchannel::Transmit()
CL:  CLoopModeLevelLoad::OnClientFrameSimulate switching to "game" loopmode with addons: 
SwitchToLoop game requested:  id [2] addons []
CL:  CNetworkGameClient already exists for connection to '=[A:1:4230360073:6362]'
CL:  connection to '=[A:1:4230360073:6362]' already started
Game: "Dota 2"
Map: "dota"
Players: 10 (1 bots) / 64 humans
Build: 7040
Server Number: 2
CL:  CNetworkGameClient::ProcessServerInfo
Initializing script VM...
...done
C:Gamerules: entering state 'DOTA_GAMERULES_STATE_INIT'
CL:  CGameRulesGameSystem::GameInit installed game rules
ReadSteamRemoteStorageFile( bufOut, "voice_ban.dt" ) -> 0.000430 seconds
UnserializeDMX: Unable to open file "0"
UnserializeDMX: Unable to open file "0"
ERROR: CParticleSnapshot::Unserialize - could not load file 0!
File error loading resource header "materials/effects/fog_world_00.vtex_c" (Error: ERROR_FILEOPEN)
File error loading resource header "materials/effects/fow_clouds_00.vtex_c" (Error: ERROR_FILEOPEN)
File error loading resource header "particles/econ/events/fall_major_2015/compendium_points_fall_ambient_2015_magic.vpcf_c" (Error: ERROR_FILEOPEN)
Telling Steam not to update the app for the next 900 seconds
CL:  CGameClientConnectPrerequisite connection succeeded
ChangeGameUIState: DOTA_GAME_UI_STATE_LOADING_SCREEN -> DOTA_GAME_UI_DOTA_INGAME
CL:  IGameSystem::LoopActivateAllSystems
HO:  IGameSystem::LoopActivateAllSystems
ReadSteamRemoteStorageFile( bufOut, "scripts/control_groups.txt" ) -> 0.028629 seconds
Remote Connect (=[A:1:4230360073:6362])
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( game : success )
ChangeGameUIState: DOTA_GAME_UI_DOTA_INGAME -> DOTA_GAME_UI_DOTA_INGAME
Telling Steam not to update the app for the next 900 seconds
CL:  Server disconnected: #GameUI_Disconnect_ServerShutdown
CL:  Disconnecting from server: #GameUI_Disconnect_Request_HSIdle
SwitchToLoop levelload requested:  id [3] addons []
Deleting lobby data from disk.
Received Stats from ord#9 (162.254.193.70:27023) when we thought we didn't have a session.  Marking session as active.
Telling Steam it is safe to udpate the app
CDOTA_SF_HTML disabled in Source 2
CSwapChainBase::QueuePresentAndWait() looped for 21 iterations without a present event.
CL:  IGameSystem::LoopDeactivateAllSystems
WriteSteamRemoteStorageFileAsync( "scripts/control_groups.txt" ) -> at 396.541
HO:  IGameSystem::LoopDeactivateAllSystems
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(dota) no such group
Unloading custom UI:
CL:  CGameRulesGameSystem::GameShutdown uninstalled game rules
CL:  CGameRules::CGameRules destructed
WriteSteamRemoteStorageFileAsync( "voice_ban.dt" ) -> at 396.858
ChangeGameUIState: DOTA_GAME_UI_DOTA_INGAME -> DOTA_GAME_UI_STATE_LOADING_SCREEN
Idle (levelload)
CAsyncWriteInProgress::OnComplete( "scripts/control_groups.txt" ) -> Success at 396.888
CAsyncWriteInProgress::OnComplete( "voice_ban.dt" ) -> Success at 396.888
CL:  CLoopModeLevelLoad::OnClientFrameSimulate switching to "game" loopmode with addons: 
SwitchToLoop game requested:  id [3] addons []
SV:  maxplayers set to 1
Initializing script VM...
...done
SV:  Spawn Server: <empty>
CL:  CWaitForGameServerStartupPrerequisite done waiting for server
CL:  CCreateGameClientJob creating client connection to 'loopback'
CL:  Sending connect to loopback
SV:  Sending S2C_CHALLENGE [2969673962 auth 2] to loopback
CL:  Received S2C_CHALLENGE [2969673962 auth 2] from loopback
CL:  Sending C2S_CONNECT [43 protocol 2969673962 auth 2] to loopback
SV:  Receiving C2S_CONNECT [protocol 43 2969673962 auth 2] from loopback
SV:  Sending S2C_CONNECTION to loopback [addons:'']
SV:  Sending S2C_CONNECTION to loopback [addons:'']
CL:  Received S2C_CONNECTION from loopback [addons:'']
CL:  Connected to 'loopback'
SV:  Sending server info to client 'I TROLL HARD' at loopback
SV:  WriteInitialSpawnGroups sending 1 groups
Game: "Dota 2"
Map: "<empty>"
Players: 1 (0 bots) / 1 humans
Build: 7040
Server Number: 2
CL:  CNetworkGameClient::ProcessServerInfo
Initializing script VM...
...done
ReadSteamRemoteStorageFile( bufOut, "voice_ban.dt" ) -> 0.000460 seconds
CL:  CGameClientConnectPrerequisite connection succeeded
ChangeGameUIState: DOTA_GAME_UI_STATE_LOADING_SCREEN -> DOTA_GAME_UI_STATE_DASHBOARD
SV:  IGameSystem::LoopActivateAllSystems
HO:  IGameSystem::LoopActivateAllSystems
SV:  Game started
CL:  IGameSystem::LoopActivateAllSystems
ReadSteamRemoteStorageFile( bufOut, "scripts/control_groups.txt" ) -> 0.134640 seconds
Failed to load grid nav data, file 'maps/<empty>.gnv' not found!
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( game : success )
SV:  Sending full update to client I TROLL HARD (reason:  initial update)
ShouldClientReceiveStringTableUserData called with NULL GameRules() on table userinfo, string 0
CL:  Receiving uncompressed update from server
CL:  Signon traffic "client":  incoming 33.586 KB [6 pkts], outgoing 1.177 KB [7 pkts]
**** Panel MapPreview has fill-parent-flow for width, but isn't in a flowing right layout
**** Panel MapPreview has fill-parent-flow for width, but isn't in a flowing right layout

```

**Comment by gdrewb-valve** [comment]
Thanks, can you post a link to your Steam community page or your Steam ID (I'm just double-checking what your Steam identity is without having you disclose anything sensitive)?

**Comment by mmuldoon** [comment]
http://steamcommunity.com/profiles/76561197974720502/

**Comment by gdrewb-valve** [comment]
The logs seem to indicate that the server just didn't get follow-up data from you after the initial few.  I know this might cause another ban (I can address that if necessary) but can you add +net_maxroutable 600 to your launch options and see if that helps?

Have you had any issues with your networking?  It appears that you've been playing some, how often does this failure happen for you?

**Comment by mmuldoon** [comment]
I haven't had issues with the network that I'm aware of, again this all seemed to start happening around the same time as the January update. I do get connected sometimes, it seems hit and miss. ISP possibly... I'll add the maxroutable to my options and let you know.

**Comment by ravzski** [comment]
I have actually solved this project accidentally.
When the "loading" page comes up after accepting a match.
I press cmd+tab, to show the steam overlay.
I then waited while the steam overlay is still on and it push thru the game.

**Comment by gdrewb-valve** [comment]
Interesting, that doesn't suggest an obvious issue as the overlay isn't supposed to interfere with the game but we'll have to see if that works for other people.

**Comment by Multisaft** [comment]
awsome, thanks mmuldoon, that does solve my issue as well:
first ever load into single draft since january update
see my topic.

**Comment by gdrewb-valve** [comment]
@mmuldoon your game wasn't actually crashing, was it?  You're welcome to try the overlay thing and if it fixes your problem that'd make it even more strange.

**Comment by mmuldoon** [comment]
Ok so the overlay trick didn't work for me, and no dota doesn't crash it just stalls out. Was able to play a few games this morning but now this afternoon had 3 stalls and abandons. @gdrewb-valve with the +net_maxroutable 600 do I just add that in the console before I start matchmaking?

**Comment by gdrewb-valve** [comment]
You would add +net_maxroutable 600 to your Steam launch options for Reborn.  You could try it from the console by doing the same thing without the + but it's simpler to put it in the launch options and then it's guaranteed to apply for the entire time you're running Reborn.

Your bans should be cleared.

**Comment by mmuldoon** [comment]
@gdrewb-valve 
Man I thought the +net was working but I'm still getting abandons every now and then. I'm locked out right now for 11 more hours. IDK if you have any other ideas or not, but I'm about at the stage of closing out this issue. Just not sure what else can be done.

**Comment by gdrewb-valve** [comment]
I'll pass this on to the networking team.  They did suggest lowering from 600 to 576, which is the absolute minimum, plus also adding

+net_showudp 1 +net_showsplits 1 +log_verbosity splitpacket detailed

to your launch options to get a lot more networking debugging spew.  If you have a problem with those options can you put up a gist of your console logs.

I can check on any bans but not until tomorrow afternoon (Feb. 12 US Pacific Time afternoon).

**Comment by gdrewb-valve** [comment]
Were you able to get any more info?

**Comment by mmuldoon** [comment]
How do I dump all the console info, it just keeps going and going. When I try and copy it, I can't get all of the info. I just got an abandon.

**Comment by gdrewb-valve** [comment]
Add -con_logfile to your launch options.  All of the console output will be saved in a file called console.log that'll be in the game/dota subdirectory of your Reborn install.

**Comment by gdrewb-valve** [comment]
Then when you post it here use the 'Gist' option at the top of the page instead of cutting and pasting all of the text into the messages.

**Comment by mmuldoon** [comment]
Well this is a new one, not sure if it's related to my current problem or because of something else. Got into a game 40min in, boots me out because of a timeout, I'm then unable to reconnect and I assume get an abandon. 
https://gist.github.com/mmuldoon/7e46cc2ec92df3de59f1

**Comment by gdrewb-valve** [comment]
Thanks, I don't see anything obvious in the log but this has gone beyond a Mac or Linux issue and my knowledge.  I've passed the log on to the larger team to see if they have any insight.

**Comment by gdrewb-valve** [comment]
Do you have a workshop_log.txt in your Steam (not Reborn) logs directory?

**Comment by mmuldoon** [comment]
workshop_log.txt
https://gist.github.com/mmuldoon/88eae5a8bc80fef6dcfe

**Comment by gdrewb-valve** [comment]
I've passed the new log on, thanks.

**Comment by mmuldoon** [comment]
Since the new update I'm getting hit with abandons again. New 24hr ban, woot! Here's my console.log
https://gist.github.com/mmuldoon/0cd2d8e500433b4cfb2561a06ee3cfb6

**Comment by gdrewb-valve** [comment]
I've passed this log on.

**Comment by gdrewb-valve** [comment]
According to the larger team your log shows network communication failures, which will prevent the game from running normally.  There's nothing in the logs to show a particular reason, just that some network traffic isn't getting through correctly.

**Comment by gdrewb-valve** [comment]
The larger team is going to continue to look at this but they don't have any ideas other than that you appear to have general networking issues.
